authentication/tests/test_authentication.py

```diff
@@ -1,23 +1,107 @@
+        "INVALID_ISSUER_TOKEN": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOjQ3MjYwMjE5ODQuODY2Mjc2LCJpc3MiOiJwaWNoZSIsImF1ZCI6InBvcnRhaWwiLCJqdGkiOiJlYzdiN2IwYWM0ZGY0YmRjODE5ZDNiNmRmYTNjZjhjYyIsInVzZXIiOiIxN3NpbXBsZSJ9.gA1u-m8b-uZlmhYrBdqo1vx1Ggic9XTN-uYiUNbSKuvrsKknE9Gn1JY24DRZjL3Xz_2zoZBAtUxwOYfJvy9GplHEsgSV0_ARDzZ4cAvWqtDpC8Eq6YusZtlN2O9PPStYYgjrgySw0bqEmUa3Vpo9PSPKEggyPq6UO-NkGbAHpLswxc8_jCwqD7795kYHZ0qIw7UKQ_DZAO_C-xnv2hWYlhUHrpk_Hy6Uteoe908ZtVAhB24ZSRor-VsXfP6UepiZ7Ehc20ggm7-mI4yyIbrA865jZldW7CqjRwYyMVoe_dbIBJ2xyzllc2gS_OfwdjcuGQRNVXcToxgvv8cv5v5IWg",
+        "INVALID_AUDIENCE_TOKEN": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOjQ3MjYwMjE5ODQuODY2Mjc2LCJpc3MiOiJzc29fc2VydmVyIiwiYXVkIjoicGljaGUiLCJqdGkiOiI4NGI1NTQzZGE2MDI0ZjU2OGE3YWEwNDJlNWJhYTZkNyIsInVzZXIiOiIxN3NpbXBsZSJ9.M7UpJQ9RwrYSsENFw_iN-ODzBPm3x90GTbjrgnGKvTxexyaA-P_KwE1cHCNdVtWqjQam-Zq88xeD_31jE1Fp7LANsQkA9ACfVuDGsOSrEcCn7yaRRJTkwxShyykI6HnNJS2MznRktOgRYUN-usJj30kNS1p7_ZQnsCYRbWt_frwztdPwlnOishTsipOBTLnmk_h9mUlS9YRNb6gsCw8FaPW3BCu7K9O2YbldpGBhq1iqPx26t2W8VeE6l7IT0soNz-KS0xGxwImuzlDQMkFxxRnnzwpyqDfXoleaZO7iTSCRNqzcae9i3fskMqeLbFrrkIlUmUZjTWz2sA-pNHDjfA",
+        "VALID_PICHE_TOKEN": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOjQ3MjYwMjE5ODQuODY2Mjc2LCJpc3MiOiJzc29fc2VydmVyIiwiYXVkIjoicG9ydGFpbCIsImp0aSI6IjE4NGM2OTJmNDBhMzQ4M2U5NGY1MDY3MzUzODBhYzIzIiwidXNlciI6InBpY2hlIn0.bNT2NkYYyrrqk551UHQd1HE02T9MRNNsc0M1Gr9bXHQv3bQBhrdTa7ILlcHf4U-rXiVKCh9QpsQZ379-jEUMFWVqSmse17z156Sdk4vQsVGILk660vOwUV_ugyQIXZCbARNTP4FhpXn4GmNMy15zOF2qn1xUw-lPgkEZnxeIuT_eDzFH_8lKjkIthdhBn7jHGiHJ-Kcu2GpOUZd6N44hD3nlAuEgVYRIN4plNFsLq6p-44eXZraF0JebjJa4J0dj-TiQ3e7XsFDjWuwd4K-bkhZF1KfeDQcFMqT7coZIqaC1o88-zQP0n0UYMtjoWVi56wT5Ams3qG_cg_6mOEFrsQ",
+        "VALID_17SIMPLE_TOKEN": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOjQ3MjYwMjE5ODQuODY2Mjc2LCJpc3MiOiJzc29fc2VydmVyIiwiYXVkIjoicG9ydGFpbCIsImp0aSI6IjJlOTEwN2ZkODc3YjRhOTA4NzdkZGI1Njc4MjczZTc1IiwidXNlciI6IjE3c2ltcGxlIn0.SaCYMt6M-KtuZQdSsKfJN9v7REpeSKJk9noFxZC5A43dyaQYuk-yiakCH_tOGmD45lvvO3mxUd9AgmvSVrDoAhU_ybujWRV30stF00j8tTaD70zncISBLM3vFCTDWpcq1JX6_iXRZsYYA_JsasY-dDwBURVS9b-YvtZj4vFSA3OF87KUmce8PNEJ9MZiuei7NkIYlKhgYXO24Y0D3HhqrFj2DD2okajU2BCsmuCfhd6lJXKsu5l4TG8jNpfc4ao6mmCtJVSOOMtpMgLts6Lg3q9rZqAF2UIYyCgrQ7OajQ2GkkurHGv_in-5FbaLiwI36Cq2XYFOlVsMgwvfw9GuLg",
+    }
+
+    def login(self, token: str = ""):
+        return self.get(f"/auth/login/?access={token}")
+
+    def logout(self):
+        return self.get(f"/auth/logout/")
+
+    def assertIsAuthenticated(self, user_id: str):
+        res = self.get("/auth/check/")
+        self.assertStatusCode(res, 200)
+        self.assertEqual(user_id, res.data["user_id"])
+
+    def assertIsNotAuthenticated(self):
+        res = self.get("/auth/check/")
+        self.assertStatusCode(res, 401)
+
+    #########
+    # LOGIN #
+    #########
+
+    def test_if_no_token_then_cannot_login(self):
+        res = self.get("/auth/login/")
+        self.assertStatusCodeIn(res, [401, 403])
```

Comment:
Why don't we know the status code ?

Reply:
This is quantum computing
(jkd, it should be a 403)

Reply:
I mean, a 401

Reply:
For some reason, raising `AuthenticationFailed` returns a 403.
Whatever.

Reply:
Then put `403` and explain it should be a `401`